libminifi/src/utils/StringUtils.cpp

```diff
@@ -602,4 +602,12 @@ std::string repeat(std::string_view str, size_t count) {
   return result;
 }
 
+std::optional<std::string> partAfterLastOccurrenceOf(std::string_view input, char delimiter) {
+  const size_t last_pos = input.find_last_of(delimiter);
+  if (last_pos == std::string::npos) {
+    return std::nullopt;
+  }
+  return std::string{input.substr(last_pos + 1)};
+}
```

Comment:
This works, but as far as I can see, in all cases where this is used, we use the full input if the return value is `nullopt`. So the code could be made simpler if the helper function returned the full input in these cases:
```suggestion
std::string partAfterLastOccurrenceOf(std::string_view input, char delimiter) {
  const size_t last_pos = input.find_last_of(delimiter);
  if (last_pos == std::string::npos) {
    return {input};
  }
  return {input.substr(last_pos + 1)};
}
```
(this will do one extra allocation in some cases, but I don't think we need to worry about that)

Reply:
Good point, I think in our use case it's a better approach, updated in https://github.com/apache/nifi-minifi-cpp/pull/1895/commits/efe1fd8a20027f98bc734747bfaa4c3376aaba5f